### ISSUE TITLE
`break-word` on inline code blocks is a bit too eager

### ISSUE THREAD

**Issue by RaceProUK** [mkdocs/mkdocs#709]
Example:
![image](https://cloud.githubusercontent.com/assets/10373748/9253910/8468db5e-41d8-11e5-9fba-5861ed8499f2.png)

I understand why it was added in response to issue #313, but I can't help but feel there's a better solution?

**Comment by d0ugal** [comment]
Hrm, yeah. That is unfortunate. I don't know a better solution tho' - one to look into.

**Comment by facelessuser** [comment]
I added some additional css that adds `white-space: pre-wrap;` fix the issue for myself:

```css
.rst-content code, .rst-content tt {
  white-space: pre-wrap;
}
```

This causes the breaking to be a bit more sane.  I don't know if there is a better way or how cross browser this is, but it seems to work for me.

**Comment by facelessuser** [comment]
FYI, out of curiosity, I just ran a test on multiple modern browsers, and it seems to behave fine in all modern browsers I tried.  I'm sure if someone is using some super old IE or something similar, it might not work, but you could say that for a lot of things.  Anyways, I would be interested to hear more people weigh in on this as my programming experience isn't really in web based programming.

**Comment by waylan** [comment]
Hmm, these look to be two different issues which are at odds with each other. Using `word-wrap: break-word;` causes the issue raised here. The solution for this issue is to have the entire code span pushed to the next line (the behavior of `word-wrap: normal`). However, to keep the solution for #313, we need  `word-wrap: break-word;` to **only** apply to code spans which are longer than the width of the parent. I'm not aware of a way to do that. In other words, we can't have both.

@facelessuser I don't see how `white-space: pre-wrap;` fixes the issue raised in #313 (or has any bearing on this issue for that matter). The fix there is meant to address a single word (text with no whitespace) which is too long to fit on one line. `white-space: pre-wrap;` has no effect on that and does not alter the behavior of `word-wrap: break-word;`.

How is this issue being handled by Sphinx themes and other documentation specific themes? I suspect MkDocs should follow their lead. My guess is the fix for #313 should get backed out, lacking any reasonable workaround.

**Comment by facelessuser** [comment]
@waylan, with both `white-space: pre-wrap;` **and** `word-wrap: break-word;` together, breaking will occur is sane places opposed to breaking a word when it doesn't need to, *but* if it has no choice, it will break a word.  I think it is completely relevant unless I am misunderstanding something.  This is what I have seen using these two together inline code blocks.

Am I missing something?

**Comment by facelessuser** [comment]
Keep in mind, what I posted above was added to, not in place of.  It was in my custom CSS that gets appended to the current CSS.

**Comment by waylan** [comment]
> @waylan, with both white-space: pre-wrap; and word-wrap: break-word; together, breaking will occur is sane places opposed to breaking a word when it doesn't need to, but if it has no choice, it will break a word. 

Maybe you are right. Actually, I just realized I can't seem to reproduce the reported issue in Chrome (and don't have access to other browsers right now). So, I can't verify if it works.

Below is an alternate fix to #313 I had typed up, which may or may not be necessary:

What if the following CSS was added to the theme:

```css
.breakword {
  word-wrap: break-word;
}
```

Then, when a user has code spans which contain a single word which is longer than the parent's width, they can apply that class to the code span.

Of course, raw HTML will always work:

```HTML
<code class="breakword">areallylongwordthatgoesonandonandonforareallylongtimeandsoonandsoforth</code>
```

Or, if they enable the [Attribute Lists][1] extension, they can use an attr_list to assign the class to the code span:

```
`areallylongwordthatgoesonandonandonforareallylongtimeandsoonandsoforth`{.breakword}
```

A third option would be to not provide any CSS for it and simply require the user to use raw HTML with inline styles defined:

```html
<code style="word-wrap: break-word;">areallylongwordthatgoesonandonandonforareallylongtimeandsoonandsoforth</code>
```

[1]: https://pythonhosted.org/Markdown/extensions/attr_list.html

**Comment by facelessuser** [comment]
> A third option would be to not provide any CSS for it and simply require the user to use raw HTML with inline styles defined:

> `<code style="word-wrap: break-word;">areallylongwordthatgoesonandonandonforareallylongtimeandsoonandsoforth</code>`

I understand the thinking behind all this, but frankly, I don't think anyone wants to have to worry about whether a line is too long or not.  They simply want it to wrap in a *reasonable* matter (maybe not perfect).  I think breaking a word like in the OP above is not reasonable.

Now, what I applied is simply a workaround to patch what I was seeing.  I'm not sure *exactly* what the root cause is.  Is it a compound issue because of the CSS of parents?  I don't really know, and at the time I was simply satisfied that the issue was cleared up.  I don't have code blocks running long in mobile browsers, and I don't have weird word breaking issues with them either.   That was enough for me, but again, I am no CSS expert.  If what I suggested is unreasonable, I feel there must be a way to achieve the same end result.
